### PR TITLE
Method to create and start task manager task

### PR DESCRIPTION
**Pull request by Deexie**
In most cases, tasks manager's tasks are started just after they are
created. Thus, to reduce boilerplate required for creating and starting
tasks, tasks::task_manager::module::make_and_start_task method is added.

Repair tasks are modified to use the method where possible.

### CODE REVIEW

**Comment by Deexie** [comment]
@bhalevy review please

File: tasks/task_manager.hh [hunk 1]
```diff
@@ -187,6 +187,19 @@ public:
         // If task has a parent, data concerning its children is updated and sequence number is inherited
         // from a parent and set. Otherwise, it must be set by caller.
         future<task_ptr> make_task(task::task_impl_ptr task_impl_ptr, task_info parent_d = task_info{});
+
+        // Must be called on target shard.
+        template<typename TaskImpl, typename... Args>
+        SEASTAR_CONCEPT( requires std::is_base_of_v<task::impl, TaskImpl> &&
```
Comment (bhalevy) [inline tasks/task_manager.hh:193]: scylla supports c++20 so no need for `SEASTAR_CONCEPT` to encapsulate the `requires` concepts.

File: tasks/task_manager.hh [hunk 2]
```diff
@@ -187,6 +187,19 @@ public:
...
+        requires (module_ptr module, Args&&... args) {
+            {TaskImpl(module, std::forward<Args>(args)...)} -> std::same_as<TaskImpl>;
+        })
+        future<task_ptr> create_and_start_task(tasks::task_info parent_info, Args&&... args) {
```
Comment (bhalevy) [inline tasks/task_manager.hh:197]: `make_and_start_task` would be better, as it calls `make_task`.
It's kind of long but I don't have a better, shorter name in mind.

**Comment by Deexie** [comment]
- create_and_start_task -> make_and_start_task
- got rid of SEASTAR CONCEPT for make_and_start_task's concept

**Review by bhalevy — APPROVED** [review]

**Review by denesb — APPROVED** [review]
Thanks, I just asked for this while reviewing the new compaction task PRs. Please use it there too.

**Pull request merged**